test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -5696,29 +5696,32 @@ BOOST_AUTO_TEST_CASE(strip_reason_strings)
 			}
 		}
 	)";
-	m_revertStrings = RevertStrings::Default;
-	compileAndRun(sourceCode, 0, "C");
+	ALSO_VIA_YUL(
+		DISABLE_EWASM_TESTRUN()
+		m_revertStrings = RevertStrings::Default;
+		compileAndRun(sourceCode, 0, "C");
 
-	if (
-		m_optimiserSettings == OptimiserSettings::minimal() ||
-		m_optimiserSettings == OptimiserSettings::none()
-	)
-		// check that the reason string IS part of the binary.
-		BOOST_CHECK(toHex(m_output).find("736f6d6520726561736f6e") != std::string::npos);
+		if (
+			m_optimiserSettings == OptimiserSettings::minimal() ||
+			m_optimiserSettings == OptimiserSettings::none()
+		)
+			// check that the reason string IS part of the binary.
+			BOOST_CHECK(toHex(m_output).find("736f6d6520726561736f6e") != std::string::npos);
 
-	m_revertStrings = RevertStrings::Strip;
-	compileAndRun(sourceCode, 0, "C");
-	// check that the reason string is NOT part of the binary.
-	BOOST_CHECK(toHex(m_output).find("736f6d6520726561736f6e") == std::string::npos);
-
-	ABI_CHECK(callContractFunction("f(bool)", true), encodeArgs(7));
-	ABI_CHECK(callContractFunction("f(bool)", false), encodeArgs());
-	ABI_CHECK(callContractFunction("g(bool)", true), encodeArgs(8));
-	ABI_CHECK(callContractFunction("g(bool)", false), encodeArgs());
-	ABI_CHECK(callContractFunction("f1(bool)", true), encodeArgs(9));
-	ABI_CHECK(callContractFunction("f1(bool)", false), encodeArgs());
-	ABI_CHECK(callContractFunction("g1(bool)", true), encodeArgs(10));
-	ABI_CHECK(callContractFunction("g1(bool)", false), encodeArgs());
+		m_revertStrings = RevertStrings::Strip;
+		compileAndRun(sourceCode, 0, "C");
+		// check that the reason string is NOT part of the binary.
+		BOOST_CHECK(toHex(m_output).find("736f6d6520726561736f6e") == std::string::npos);
```

Comment:
Ah it has to be endtoendtest because of this? Can't we just have a commandline test, that's where we keep bytecode checks right now.

Reply:
I would prefer not to. For commandline tests, I'm always worried that they miss the point because someone just updates without thinking about what the test does.

Reply:
I meant to have semantic tests for actual outputs, and a cli test for the bytecode.

But it would be better to support checking the bytecode in semantic tests.

Reply:
Here's an idea #10820.